File: server/06-unusual-spending/unusual-spending.js
import {fetch} from './fetch';
import {categorize} from './categorize';
import {email} from './email';

const unusualSpending = (userId) => {
  const payments = fetch(userId);
  const categorizedPayments = categorize(payments);
  email(userId, categorizedPayments);
};

export {unusualSpending};

File: unusual-spending/src/main/java/spending/UnusualSpending.java
package spending;

class UnusualSpending {

  private
  Fetcher fetcher;

  private
  Categorizer categorizer;

  private
  Emailer emailer;

	void sendEmail(long userId) {
    Payments payments = fetcher.fetch(userId);
    Payments categorized = categorizer.categorize(payments);
    emailer.email(userId, categorized);
  }

}
